src/main/kotlin/org/pkl/lsp/completion/UnqualifiedAccessCompletionProvider.kt

```diff
@@ -218,6 +238,18 @@ class UnqualifiedAccessCompletionProvider(private val project: Project) : Comple
     return isClassOrTypeAlias(type)
   }
 
+  private fun editSource(source: String, line: Int, col: Int, edit: String): String {
+    return source
+      .lines()
+      .mapIndexed { i, txt ->
+        if (i == line) {
+          val end = col.coerceAtMost(txt.length)
+          txt.substring(0, end) + edit + txt.substring(end)
+        } else txt
+      }
+      .joinToString("\n")
+  }
```

Comment:
I think let's move this to utils so that we can also use this in `QualifiedAccessCompletionProvider`